web/impact/impact/views/algolia_api_key_view.py

```diff
@@ -58,20 +62,22 @@ def get(self, request, format=None):
 
 
 def _get_search_key(request):
-    if request.user.is_staff:
+    if is_employee(request.user):
         return settings.ALGOLIA_STAFF_SEARCH_ONLY_API_KEY
     return settings.ALGOLIA_SEARCH_ONLY_API_KEY
 
 
 def _get_filters(request):
-    if request.user.is_staff:
+
+    # an empty filter i.e. [], means the user sees all mentors
+    if is_employee(request.user):
         return []
     participant_roles = [UserRole.AIR, UserRole.STAFF, UserRole.MENTOR]
 
-    participant_roles = _entrepreneur_specific_alumni_filter(
```

Comment:
this change means we're beating up on the `participant_roles` list in the function and implicitly changing its state in a way that's not obvious at the point of call. This seems to me like a loss of clarity. If we want to take this route, we should make the statefulness explicit by making it a field of the object, and make the filter functions into instance methods.

Reply:
I understand your point, I'll revert.